src/libraries/System.Private.CoreLib/src/System/Threading/Tasks/TaskCompletionSource.cs

```diff
@@ -32,332 +23,268 @@ namespace System.Threading.Tasks
+            }
 
             bool rval = _task.TrySetException(exception);
-            if (!rval && !_task.IsCompleted) SpinUntilCompleted();
+            if (!rval && !_task.IsCompleted)
+            {
+                _task.SpinUntilCompleted();
+            }
+
             return rval;
         }
 
         /// <summary>
-        /// Attempts to transition the underlying
-        /// <see cref="System.Threading.Tasks.Task{TResult}"/> into the
-        /// <see cref="System.Threading.Tasks.TaskStatus.Faulted">Faulted</see>
-        /// state.
+        /// Attempts to transition the underlying <see cref="Tasks.Task"/> into the <see cref="TaskStatus.Faulted"/> state.
         /// </summary>
-        /// <param name="exceptions">The collection of exceptions to bind to this <see
-        /// cref="System.Threading.Tasks.Task{TResult}"/>.</param>
+        /// <param name="exceptions">The collection of exceptions to bind to this <see cref="Tasks.Task"/>.</param>
         /// <returns>True if the operation was successful; otherwise, false.</returns>
-        /// <remarks>This operation will return false if the
-        /// <see cref="System.Threading.Tasks.Task{TResult}"/> is already in one
-        /// of the three final states:
-        /// <see cref="System.Threading.Tasks.TaskStatus.RanToCompletion">RanToCompletion</see>,
-        /// <see cref="System.Threading.Tasks.TaskStatus.Faulted">Faulted</see>, or
-        /// <see cref="System.Threading.Tasks.TaskStatus.Canceled">Canceled</see>.
+        /// <remarks>
+        /// This operation will return false if the <see cref="Tasks.Task"/> is already in one of the three final states:
+        /// <see cref="TaskStatus.RanToCompletion"/>,
+        /// <see cref="TaskStatus.Faulted"/>, or
+        /// <see cref="TaskStatus.Canceled"/>.
         /// </remarks>
-        /// <exception cref="System.ArgumentNullException">The <paramref name="exceptions"/> argument is null.</exception>
-        /// <exception cref="System.ArgumentException">There are one or more null elements in <paramref name="exceptions"/>.</exception>
-        /// <exception cref="System.ArgumentException">The <paramref name="exceptions"/> collection is empty.</exception>
-        /// <exception cref="System.ObjectDisposedException">The <see cref="Task"/> was disposed.</exception>
+        /// <exception cref="ArgumentNullException">The <paramref name="exceptions"/> argument is null.</exception>
+        /// <exception cref="ArgumentException">There are one or more null elements in <paramref name="exceptions"/>.</exception>
+        /// <exception cref="ArgumentException">The <paramref name="exceptions"/> collection is empty.</exception>
         public bool TrySetException(IEnumerable<Exception> exceptions)
         {
-            if (exceptions == null) ThrowHelper.ThrowArgumentNullException(ExceptionArgument.exceptions);
+            if (exceptions is null)
```

Comment:
> is nu [](start = 26, length = 6)

just curios, why prefer is than == here?

Reply:
No good reason here, otherwise it was the mindset I was in :)  The general benefit of `is` over `==` is that the former doesn't factor in operator==, so it's a pure null check, but that's not relevant here, as there's no operator== on IEnumerable.